analyzers/tests/SonarAnalyzer.Test/Rules/LoggingTemplatePlaceHoldersShouldBeInOrderTest.cs

```diff
@@ -0,0 +1,133 @@
+﻿/*
+* SonarAnalyzer for .NET
+* Copyright (C) 2015-2024 SonarSource SA
+* mailto: contact AT sonarsource DOT com
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+using SonarAnalyzer.Rules.CSharp;
+using SonarAnalyzer.Rules.MessageTemplates;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class LoggingTemplatePlaceHoldersShouldBeInOrderTest
+{
+    private static readonly IEnumerable<MetadataReference> LoggingReferences =
+        NuGetMetadataReference.MicrosoftExtensionsLoggingAbstractions()
+        .Concat(NuGetMetadataReference.NLog(Constants.NuGetLatestVersion))
+        .Concat(NuGetMetadataReference.Serilog(Constants.NuGetLatestVersion));
```

Comment:
I would add each reference to each test here, to be more precise with the dependencies.
It will also make it easier to find out issues with the conditional `CompilationStart` registration, if an assembly we are searching gets out of sync.